### PR TITLE
update kubevirt config,template,job

### PR DESCRIPTION
**Pull request by sallyom**
* template is cluster-launch-installer-custom-test-image.yaml
* requires https://github.com/kubevirt/kubevirt/pull/2991

### CODE REVIEW

File: ci-operator/templates/openshift/installer/cluster-launch-installer-nested-virt-tests.yaml [hunk 1]
```diff
@@ -104,264 +104,114 @@ objects:
...
+        # set up cloud-provider-specific env vars
+        export KUBE_SSH_BASTION="$( oc --insecure-skip-tls-verify get node -l node-role.kubernetes.io/master -o 'jsonpath={.items[0].status.addresses[?(@.type=="ExternalIP")].address}' ):22"
+        export KUBE_SSH_KEY_PATH=/tmp/cluster/ssh-privatekey
+        if [[ "${CLUSTER_TYPE}" == "gcp" ]]; then
+          export GOOGLE_APPLICATION_CREDENTIALS="${GCP_SHARED_CREDENTIALS_FILE}"
+          export KUBE_SSH_USER=core
+          mkdir -p ~/.ssh
+          cp /tmp/cluster/ssh-privatekey ~/.ssh/google_compute_engine || true
+          export TEST_PROVIDER='{"type":"gce","region":"us-east1","multizone": true,"multimaster":true,"projectid":"openshift-gce-devel-ci"}'
+        elif [[ "${CLUSTER_TYPE}" == "aws" ]]; then
```
Comment (dhiller) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-nested-virt-tests.yaml]: To my knowledge nested virtualization on AWS is supported only with emulation and that should be dead slow? Am I wrong here?

**Review by dhiller — COMMENTED** [review]
Could we for the moment set `CLUSTER_TYPE` to `gcp`, so that we skip all the other cluster types in testing? Maybe we can enable this in a later step?

File: ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml [hunk 2]
```diff
@@ -5,9 +5,17 @@ base_images:
     tag: base
 build_root:
   project_image:
-    context_dir: hack/kubevirt-builder
-    dockerfile_path: Dockerfile
+    dockerfile_path: hack/kubevirt-builder/Dockerfile.ci
 canonical_go_repository: kubevirt.io/kubevirt
+images:
```
Comment (dhiller) [inline ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml:12]: So you are not using the `src` image above any more, right?

**Review by dhiller — COMMENTED** [review]
Maybe using `LOCAL_IMAGE_SRC` instead of `LOCAL_IMAGE_KUBEVIRT_TEST` would work?

**Comment by dhiller** [comment]
One more ask: can we attach a secret via a volume mount also? This would be needed for accessing a quay repository for private repository access

File: ci-operator/templates/openshift/installer/cluster-launch-installer-nested-virt-tests.yaml (hunk 1 above)
Comment (sallyom) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-nested-virt-tests.yaml]: right, hm.. but, the test container is running in gcp (I believe the api.ci cluster runs in gce?) and that is what has to have nested-virt enabled, right? or does the external cluster require nested-virt?  anyways, we can switch to only test in gcp, but I was thinking you'd want to eventually test kubevirt against clusters in all supported cloud providers?

File: ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml (hunk 2 above)
Comment (sallyom) [reply ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml:12]: right, still working that out.. i'm going to have to ask testplatform team to clarify this.

**Comment by sallyom** [comment]
> One more ask: can we attach a secret via a volume mount also? This would be needed for accessing a quay repository for private repository access

yes, testplatform team can help w/ that when we get to that point

**Comment by sallyom** [comment]
getting this error during ./hack/build-manifests.sh `go build`:
`mkdir /go/pkg/mod: permission denied`  
~and there's no 'real user' nor privileged user in an OpenShift pod, so no `sudo`.  The more I dig in here, the more I think your best option is to launch the tests from a separate VM (gcp)
I'm trying to chmod 777 /go in the test image and/or adding `/go` as a volume-mount :woman_shrugging: that will get past the go build issue i think.~

EDIT: performing the builds within the test_image build now, so this is no longer an issue

**Comment by sallyom** [comment]
the tests are running! (and failing)!

File: ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml (hunk 2 above)
Comment (sallyom) [reply ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml:12]: The LOCAL_IMAGE_KUBEVIRT_TEST is the image built from this PR's Dockerfile: https://github.com/kubevirt/kubevirt/pull/2991  
Is there another image you were using b4?

Comment (sallyom) [reply ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml:12]: The custom-test-image template calls for a test-image variable, I'm using the env var 
`TEST_IMAGESTREAM_TAG=stable:kubevirt-test`

**Review by dhiller — COMMENTED** [review]
I still don't get how you managed to enable the nested virtualization? I suppose you use TEST_OPTIONS somehow but where?

**Comment by sallyom** [comment]
This is the failure now:
```
Failure [160.317 seconds]
[BeforeSuite] BeforeSuite 
/tmp/kubevirt/tests/tests_suite_test.go:48

  Timed out after 120.000s.
  Both KVM devices and vhost-net devices are required for testing, but are not present on cluster nodes
  Expected
      <bool>: false
  to be true

  /tmp/kubevirt/tests/utils.go:741
------------------------------
```
So, is that a bad error msg and should it instead read `but are not present on the host running the tests`? and if so, this means we need to run the tests from within a  GCP nested-virt-enabled VM after all (instead of within the test container in the pod)? OR, is there something that has to be configured on the OpenShift cluster to enable kubevirt properly? If so, what? How/when does this test suite run successfully? :woman_shrugging:   The setup is good, tests run, but they fail - if we need to run tests from an external gcp VM, we'll have to setup a new template for that.

**Pull request merged**